tests/test_precovery_index.py

```diff
@@ -58,27 +50,51 @@ def test_precovery(test_db_dir):
         )
         orbits_keplerian.append(orbit)
 
-    # Load observations from csv file
-    observations_df = pd.read_csv(TEST_OBSERVATION_FILE, float_precision="round_trip")
-
-    # Test that the number of frames is equal to the number of observations
-    # Note that this will only be true for a small enough number of objects that are not near
-    # each other on the sky, which is fine for our test data set
-    con = sql.connect(os.path.join(test_db_dir, "index.db"))
-    frames = pd.read_sql("""SELECT * FROM frames""", con)
-    assert len(frames) == len(observations_df)
-    con.close()
+    # Load observations from csv files and index each one (one observation file
+    # per dataset)
+    observation_files = glob.glob(
+        os.path.join(TEST_OBSERVATIONS_DIR, "dataset_*", "*.csv")
+    )
+    observations_dfs = []
+    for observation_file in observation_files:
+        observations_df_i = pd.read_csv(
+            observation_file,
+            float_precision="round_trip",
+            dtype={
+                "dataset_id": str,
+                "observatory_code": str,
+                "filter": str,
+                "exposure_duration": np.float64,
+            },
+        )
+        observations_dfs.append(observations_df_i)
+
+        dataset_id = observations_df_i["dataset_id"].values[0]
+
+        index(
+            out_dir=test_db_dir,
+            dataset_id=dataset_id,
+            dataset_name=dataset_id,
+            data_dir=os.path.join(
+                os.path.dirname(__file__), f"data/index/{dataset_id}/"
+            ),
+            nside=16,
+            cpu_count=1,
```

Comment:
`cpu_count=1` is a residual from when I disabled multi-processing during testing. I'll remove this before merging and set it back to default.